src/read.jl

```diff
@@ -171,6 +171,23 @@ function read!(buf, pos, len, b, tape, tapeidx, ::Type{String})
         b = getbyte(buf, pos)
     end
     @check
+    if inf_mapping !== nothing
+        val = view(buf, strpos:pos-1)
+        float = if val == codeunits(inf_mapping(Inf))[2:end-1]
+            Inf
+        elseif val == codeunits(inf_mapping(-Inf))[2:end-1]
+            -Inf
+        elseif val == codeunits(inf_mapping(NaN))[2:end-1]
```

Comment:
This usage makes me think that `inf_mapping` should be a Tuple or NamedTuple rather than a function.

Reply:
I tried arrays, tuples and functions.

Reply:
I also checked the RawType approach but I couldn't find out how to change the type to Float. The current approach looks more natural to me and has less code.
It is somewhat of a restriction that I only support the case of string mappings and the checking of `codeunits(...)[2:end-1]` is not completely correct and would also match mappings that are not json strings, e.g. `"_Infinity_"` instead of `"\"Infinity\""`. I just omitted further checks for performance reasons.

Reply:
The difference between function and tuple that is giving you that performance difference is methods are specialized on a function but not on a tuple's value. You could get similar performance with a tuple by lifting it to the type domain with `Val`. I do see the advantage in terms of runtime performance of having the serialization format of inf and nan be passed into write/read at the type level